app/views/posts/show.html.erb

```diff
@@ -1 +1,38 @@
-<p>Details of Posts</p>
\ No newline at end of file
+<!-- Heading -->
+<h1 class="Head">Shows the details of posts for a given user</h1>
+
+<section class="posts_container">
+  <% if @post %>
+    <!-- Post details -->
+    <div class='post'>
+      <div class=top-post>
+        <h2><%= @post.title %> by <%= @post.author.name %></h2>
+        <p class="top_comments_likes">
+          Comments: <%= @post.comments_counter %>,
+          Likes: <%= @post.likes_counter %>
+        </p>
+      </div>
+      <div>
+        <p class="post_text"><%= @post.text %></p>
+      </div>
+    </div>
+  <% else %>
+    <!-- No recent posts message -->
+    <p>No recent posts.</p>
+  <% end %>
+</section>
+
+<section class='comments-area'>
+<% if @post_comments && @post_comments.any? %>
```

Comment:
- [ ] Due to comments creating issue in your project, I was not able to test if your comments are displaying or not. Please fix it or give proper guidence how to creat posts in `rails c` to reviewers. Thank you!